pkgs/applications/virtualization/looking-glass-client/default.nix

```
@@ -43,35 +45,39 @@ let
     type = "Application";
     exec = "looking-glass-client";
     icon = "lg-logo";
-    terminal = true;
   };
 in
 stdenv.mkDerivation rec {
   pname = "looking-glass-client";
-  version = "B6";
+  version = "B7-rc1";
 
   src = fetchFromGitHub {
     owner = "gnif";
     repo = "LookingGlass";
     rev = version;
-    sha256 = "sha256-6vYbNmNJBCoU23nVculac24tHqH7F4AZVftIjL93WJU=";
+    sha256 = "sha256-ne1Q+67+P8RHcTsqdiSSwkFf0g3pSNT91WN/lsSzssU=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # Fix failing cmake assertion when disabling X11 whithout explicitly enabling Wayland.
+    ./0001-client-cmake-move-X11-config-directives-to-displayse.patch
+  ];
```

where does this patch come from, did you write it yourself? If that's the case were there any efforts made upstreaming it?

---

https://github.com/gnif/LookingGlass/commit/20972cfd9b940fddf9e7f3d2887a271d16398979

---

thank you, may I suggest dropping the patch and changing to:
```suggestion
  patches = [
    # Fix failing cmake assertion when disabling X11 whithout explicitly enabling Wayland.
    # Can be removed after the next update.
    (fetchpatch {
      url = "https://github.com/gnif/LookingGlass/commit/20972cfd9b940fddf9e7f3d2887a271d16398979.patch";
      hash = "...";
    })
  ];
```
using fetchpatch is common practice in nixpkgs because it reduces our tarball and git history size while also making the origin of patches more transparent.

---

That's what I tried at first, but I couldn't get the patch to apply without creating it with the `--no-prefix` option which GitHub doesn't seem to do on its own.

---

In my local package I do `cd client` in `postPatch` and that allows me to use normal patches

---

[fetchpatch](https://nixos.org/manual/nixpkgs/stable/#fetchpatch) has a few options to do things like that, I believe you can achieve the cd with `relative`.

---

I already played around with the available options before, but I couldn't get it to apply this way. Maybe I did something wrong though